api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -206,10 +206,9 @@ type NodePoolSpec struct {
 	// arch is the preferred processor architecture for the NodePool. Different platforms might have different supported architectures.
 	// TODO: This is set as optional to prevent validation from failing due to a limitation on client side validation with open API machinery:
 	//	https://github.com/kubernetes/kubernetes/issues/108768#issuecomment-1253912215
-	// TODO Add s390x to enum validation once the architecture is supported
 	//
 	// +kubebuilder:default:=amd64
-	// +kubebuilder:validation:Enum=arm64;amd64;ppc64le
+	// +kubebuilder:validation:Enum=arm64;amd64;ppc64le;s390x
```

Comment:
this impacts all platforms, can you please add CEL validation for this input which disambiguate by platform?

Reply:
https://github.com/openshift/hypershift/blob/main/api/hypershift/v1beta1/nodepool_types.go#L87